bdio2/src/main/java/com/blackducksoftware/bdio2/BdioMetadata.java

```diff
@@ -279,8 +299,8 @@ public BdioMetadata buildNumber(@Nullable String buildNumber) {
     /**
      * Sets the scan type
      */
-    public BdioMetadata scanType(@Nullable String scanType) {
-        putFieldValue(Bdio.DataProperty.scanType, scanType);
+    public BdioMetadata scanType(@Nullable Bdio.ScanType scanType) {
+        put(JsonLdConsts.TYPE, scanType != null ? scanType.getValue() : null);
```

Comment:
Would recommend adding Javadoc as to why this is nullable.

Reply:
Going further, this value cannot be Null. We can throw an exception if we see this value as null. As that is the actual requirement.

Reply:
Are we able to handle clients producing and uploading old documents where this is not defined?

Reply:
That would be an interesting case. If we say that, yes we want to support clients uploading the old versioned documents then we might have to up-convert the document to the latest version. And with the up-conversion process, we would have to figure out and add the scanType field in the document.

Reply:
older versioned documents are now when converted to latest version will include the scan type field.